packages/babel-parser/src/parser/statement.js

```diff
@@ -1857,10 +1857,24 @@ export default class StatementParser extends ExpressionParser {
     }
 
     const next = this.nextTokenStart();
-    return (
+    const hasFrom = this.isUnparsedContextual(next, "from");
+    if (
       this.input.charCodeAt(next) === charCodes.comma ||
-      this.isUnparsedContextual(next, "from")
-    );
+      (this.match(tt.name) && hasFrom)
+    ) {
+      return true;
+    }
+    // lookahead again when `export default from` is seen
+    if (this.match(tt._default) && hasFrom) {
+      const nextAfterFrom = this.input.charCodeAt(
+        this.nextTokenStartSince(next + 4),
+      );
+      return (
+        nextAfterFrom === charCodes.quotationMark ||
+        nextAfterFrom === charCodes.apostrophe
```

Comment:
I don't understand why we need to special-case these two characters and why not, for example `(` or `*`.

Reply:
If we see a string literal after from, it is considered a fromClause and thus it is indeed export default fromClause instead of export default AssignmentExpression.

Reply:
I was reading quaotationMark as questionMark 😁